packages/dds/tree/src/test/simple-tree/benchmarkUtilities.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import {
+	SchemaFactory,
+	type NodeFromSchema,
+	type ValidateRecursiveSchema,
+} from "../../simple-tree/index.js";
+import { hydrate } from "./utils.js";
+import { Tree } from "../../shared-tree/index.js";
+
+const schemaFactory = new SchemaFactory("test");
+
+/**
+ * A recursive tree node with a single leaf value
+ */
+export class DeepTreeNode extends schemaFactory.objectRecursive("deep-tree-node", {
+	node: [() => DeepTreeNode, schemaFactory.number],
+}) {}
+{
+	type _check = ValidateRecursiveSchema<typeof DeepTreeNode>;
+}
+
+/**
+ * A wide tree node with all leaves having the same value
+ */
+export const WideTreeNode = schemaFactory.array(schemaFactory.number);
+export type WideTreeNode = NodeFromSchema<typeof WideTreeNode>;
+
+/**
+ * Make a deep tree with a single leaf value
+ */
+function makeDeepTree(depth: number, leafValue: number): DeepTreeNode {
```

Comment:
The Unhydrated type can be used as documentation to indicate if a tree has been hydrated or not.

```suggestion
function makeDeepTree(depth: number, leafValue: number): Unhydrated<DeepTreeNode> {
```

Reply:
Updated